pkg/storage/stores/shipper/uploads/table.go

```diff
@@ -181,13 +188,28 @@ func (lt *Table) RemoveDB(name string) error {
 }
 
 // Upload uploads all the dbs which are never uploaded or have been modified since the last batch was uploaded.
-func (lt *Table) Upload(ctx context.Context) error {
+func (lt *Table) Upload(ctx context.Context, force bool) error {
 	lt.dbsMtx.RLock()
 	defer lt.dbsMtx.RUnlock()
 
+	// upload files excluding active shard. It could so happen that we just started a new shard but the file for last shard is still being updated due to pending writes or pending flush to disk.
+	// To avoid uploading it, excluding previous active shard as well if it has been not more than a minute since it became inactive.
+	uploadShardsBefore := fmt.Sprint(time.Now().Add(-time.Minute).Truncate(shardDBsByDuration).Unix())
+
+	// Adding check for considering only files which are shared and have just an epoch in their name.
+	// We can remove this check after a release or two when we will have only files with epoch in their name.
+	filenameWithEpochRe, err := regexp.Compile(`^[0-9]{10}$`)
```

Comment:
I think we can compile this once and just re-use it?

Reply:
I'm also really confused by this, I don't think this matches the old or new files?

example from one of my older Loki's running shipper: `bde91fc8191e-1588020045`

Newer: `winbookpro-1597004227840461700-1597004100`

Or maybe i didn't trace the origin of name back far enough?

Reply:
> I think we can compile this once and just re-use it?
We will have to add it as a member of `Table` type. What do you think? 

> I'm also really confused by this, I don't think this matches the old or new files?
I guess you checked this in object store. This code works on files stored in ingester's active index directory which only have a timestamp after we introduced sharding every 15 mins.

Reply:
> We will have to add it as a member of Table type. What do you think? 
we could make it global and use `MustCompile`

Could you also update the comment to include examples of the previous and current file names to see what the regex should not match and should match?

Reply:
I have updated the comment. Thanks!